### ISSUE TITLE
Tribler 7 stops working after one torrent gives 'File not found in torrent' error

### ISSUE THREAD

**Issue by pimlie** [Tribler/tribler#2808]
##### Tribler version/branch+revision: #####
```
~/bin/tribler $ git show
commit 02b0cfa8a1556f2017b50157214c50ec123a9b90
Merge: f7779ba eaec7bb
Author: Martijn de Vos
Date:   Wed Feb 15 09:40:31 2017 +0100

    Merge pull request #2788 from qstokkink/fix_socks5_associate
    
    READY: Allow UDP associate without circuits

~/bin/tribler $ git status
On branch devel
Your branch is up-to-date with 'origin/devel'.
nothing to commit, working directory clean
```
##### Operating system and version: #####
```
~/bin/tribler $ uname -a
Linux 4.4.0-53-generic #74-Ubuntu SMP Fri Dec 2 15:59:10 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```
##### Expected behavior: #####
Tribler skips file, ignores error or removes torrent from download queue if it can't continue.

##### Actual behavior: #####
Tribler hangs/crashes and after closing Tribler also hangs/crashes during startup. Replacing the `return ValueError` in `Tribler\Core\TorrentDef.py:767` with `return i` skips this error and the torrent (seems to) start to download normally

##### Steps to reproduce the behavior: #####
Add the following magnet link to Tribler with 'Import torrent from URL':

##### Relevant log file output: #####
```
ERROR:RESTRequest:[Failure instance: Traceback: <type 'exceptions.TypeError'>: ValueError('File not found in torrent',) is not JSON serializable
/usr/lib/python2.7/dist-packages/twisted/protocols/basic.py:571:dataReceived
/usr/lib/python2.7/dist-packages/twisted/web/http.py:1657:lineReceived
/usr/lib/python2.7/dist-packages/twisted/web/http.py:1736:allContentReceived
/usr/lib/python2.7/dist-packages/twisted/web/http.py:762:requestReceived
--- <exception caught here> ---
/usr/lib/python2.7/dist-packages/twisted/web/server.py:183:process
/usr/lib/python2.7/dist-packages/twisted/web/server.py:234:render
/usr/lib/python2.7/dist-packages/twisted/web/resource.py:250:render
~/bin/tribler/Tribler/Core/Modules/restapi/downloads_endpoint.py:225:render_GET
/usr/lib/python2.7/json/__init__.py:244:dumps
/usr/lib/python2.7/json/encoder.py:207:encode
/usr/lib/python2.7/json/encoder.py:270:iterencode
/usr/lib/python2.7/json/encoder.py:184:default
]
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/twisted/web/server.py", line 183, in process
    self.render(resrc)
  File "/usr/lib/python2.7/dist-packages/twisted/web/server.py", line 234, in render
    body = resrc.render(self)
  File "/usr/lib/python2.7/dist-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "~/bin/tribler/Tribler/Core/Modules/restapi/downloads_endpoint.py", line 225, in render_GET
    return json.dumps({"downloads": downloads_json})
  File "/usr/lib/python2.7/json/__init__.py", line 244, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: ValueError('File not found in torrent',) is not JSON serializable```

**Comment by devos50** [comment]
Thanks for the report!

**Comment by devos50** [comment]
I'm unable to reproduce this. Does it happen consistently? And did you select all files in the torrent?

**Comment by pimlie** [comment]
It happens consistently, downloading all files with default anonymity.

That said, is there maybe a caching mechanism for processed torrents of magnet links? I use the GIT version of Tribler so I often pull new commits. And I have loaded this magnet link weeks ago after which it failed with the above message. Yesterday I tried it again twice but it failed with the above mesage. Maybe an old cache of this torrent file keeps triggering this issue?

**Comment by devos50** [comment]
Thank you for providing more information. After some extensive debugging, we found out that the torrent file itself is inconsistent regarding some metadata which causes that one specific file cannot be found inside the torrent (in fact, we discovered an additional error when we were trying to investigate this).

Since this might affect more torrents, we will prioritise this issue. Fortunately, it is very easy to reproduce since we have the torrent that causes this error.
